### PR TITLE
Fix timezone handling

### PR DESCRIPTION
**Pull request by rs-johnwhyte**
Updated root certificate and moved to cert.h
Updated factory.ino to use new cert.h
Updated includes to newer versions
Ported setTimezone from Lilygo T-Display-S3 to handle CUSTOM_TIMEZONE
Added test_screen

### CODE REVIEW

File: .gitignore
Comment (rs-johnwhyte) [inline .gitignore:1]: Can be ignored

File: examples/factory/cert.h
Comment (rs-johnwhyte) [inline examples/factory/cert.h:1]: Should be added

File: examples/factory/factory.ino
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:1]: Should be added

File: examples/factory/factory.ino [hunk 1]
```diff
@@ -5,10 +5,12 @@
 #include "rm67162.h"
 #include "setup_img.h"
 #include "WiFi.h"
-#include "sntp.h"
+#include "esp_sntp.h"
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:8]: esp_sntp.h is the updated version of sntp.h library

File: examples/factory/factory.ino [hunk 2]
```diff
@@ -5,10 +5,12 @@
...
 #include "time.h"
 #include "factory_gui.h"
 #include "zones.h"
+#include "pins_config.h"
+#include "cert.h"
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:13]: Includes new certificate for Root

File: examples/factory/factory.ino [hunk 3]
```diff
@@ -5,10 +5,12 @@
 #include "rm67162.h"
 #include "setup_img.h"
 #include "WiFi.h"
-#include "sntp.h"
+#include "esp_sntp.h"
 #include "time.h"
 #include "factory_gui.h"
 #include "zones.h"
+#include "pins_config.h"
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:12]: Includes additional pins / configuration details for sketch

File: examples/factory/factory.ino [hunk 4]
```diff
@@ -78,34 +80,13 @@ void setup()
...
-                lv_obj_del(colors_obj);
-                break;
-            }
-        }
-        lv_timer_handler();
-        button1.tick();
-    }
-    //test color end
-
+    // test_screen(); // uncomment if you wish to test for bad pixels before factory demo.
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:83]: test_screen() is now a function that can be easily commented in / out for testing

File: examples/factory/factory.ino [hunk 5]
```diff
@@ -78,34 +80,13 @@ void setup()
...
     wifi_test();
+    LV_DELAY(2000);
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:85]: Moved these sections of code out of wifi_test() and into setup()

File: examples/factory/factory.ino [hunk 6]
```diff
@@ -78,34 +80,13 @@ void setup()
...
+    setTimezone();
+    printLocalTime();
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:87]: included printLocalTime() to assist with serial debugging on initial startup

File: examples/factory/factory.ino [hunk 7]
```diff
@@ -78,34 +80,13 @@ void setup()
 
 
 
-    //Test screen bad pixels
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:81]: Moved this section to a function test_screen();

File: examples/factory/factory.ino [hunk 8]
```diff
@@ -141,6 +120,36 @@ void loop()
     }
 }
 
+//Test screen bad pixels
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:123]: New function for test_screen() created using code from line 81 in factory.ino

File: examples/factory/factory.ino [hunk 9]
```diff
@@ -246,9 +255,6 @@ void wifi_test(void)
         Serial.println(" millseconds");
         lv_label_set_text(log_label, text.c_str());
     }
-    LV_DELAY(2000);
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:249]: Moved this code out of wifi_test() and into setup()

File: examples/factory/factory.ino [hunk 10]
```diff
@@ -270,10 +277,18 @@ void timeavailable(struct timeval *t)
 
 void setTimezone()
 {
+    Serial.println ("Setting timezone");
```
Comment (rs-johnwhyte) [inline examples/factory/factory.ino:280]: updated setTimezone() from T-Display-S3/examples/factory.ino

File: examples/factory/pins_config.h [hunk 11]
```diff
@@ -11,7 +11,7 @@
 #define EXAMPLE_LCD_V_RES     240
 #define LVGL_LCD_BUF_SIZE     (EXAMPLE_LCD_H_RES * EXAMPLE_LCD_V_RES)
 
-#define WIFI_SSID             "YOUR-SSID"
+#define WIFI_SSID             "YOUR-WIFI"
```
Comment (rs-johnwhyte) [inline examples/factory/pins_config.h:14]: can be ignored

**Comment by lewisxhe** [comment]
I have completed the certificate update. This PR will not be merged because there is a compilation error.

**Pull request closed without merging**